pkg/providers/snow/sync.go

```diff
@@ -0,0 +1,174 @@
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	snowv1 "github.com/aws/eks-anywhere/pkg/providers/snow/api/v1beta1"
+)
+
+const (
+	SnowClusterKind         = "AWSSnowCluster"
+	SnowMachineTemplateKind = "AWSSnowMachineTemplate"
+)
+
+func CAPICluster(clusterSpec *cluster.Spec) *clusterv1.Cluster {
+	cluster := clusterapi.Cluster(clusterSpec)
+	cluster.Spec.InfrastructureRef.Kind = SnowClusterKind
+	cluster.Spec.InfrastructureRef.Name = clusterSpec.GetName() // TODO: use capinamegenerator
+	return cluster
+}
+
+func KubeadmControlPlane(clusterSpec *cluster.Spec) *controlplanev1.KubeadmControlPlane {
+	kcp := clusterapi.KubeadmControlPlane(clusterSpec)
+
+	kcp.Spec.MachineTemplate.InfrastructureRef.Kind = "AWSSnowMachineTemplate"
+
+	// TODO: support unstacked etcd
+	kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.Etcd = bootstrapv1.Etcd{
+		Local: &bootstrapv1.LocalEtcd{
+			ImageMeta: bootstrapv1.ImageMeta{
+				ImageRepository: clusterSpec.VersionsBundle.KubeDistro.Etcd.Repository,
+				ImageTag:        clusterSpec.VersionsBundle.KubeDistro.Etcd.Tag,
+			},
+			ExtraArgs: map[string]string{
+				"listen-peer-urls":   "https://0.0.0.0:2380",
+				"listen-client-urls": "https://0.0.0.0:2379",
+			},
+		},
+	}
+
+	kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.APIServer = bootstrapv1.APIServer{
+		ControlPlaneComponent: bootstrapv1.ControlPlaneComponent{
+			ExtraArgs: map[string]string{
+				"cloud-provider": "external",
+			},
+		},
+	}
+
+	kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.ControllerManager = bootstrapv1.ControlPlaneComponent{
+		ExtraArgs: map[string]string{
+			"cloud-provider": "external",
+		},
+	}
+
+	kcp.Spec.KubeadmConfigSpec.InitConfiguration.NodeRegistration.KubeletExtraArgs = map[string]string{
+		"cloud-provider": "external",
+		"provider-id":    "aws-snow:////'{{ ds.meta_data.instance_id }}'",
+	}
+
+	kcp.Spec.KubeadmConfigSpec.JoinConfiguration.NodeRegistration.KubeletExtraArgs = map[string]string{
+		"cloud-provider": "external",
+		"provider-id":    "aws-snow:////'{{ ds.meta_data.instance_id }}'",
+	}
+
+	kcp.Spec.KubeadmConfigSpec.PreKubeadmCommands = []string{
+		fmt.Sprintf("/etc/eks/bootstrap.sh %s %s", clusterSpec.VersionsBundle.Snow.KubeVip.VersionedImage(), clusterSpec.Spec.ControlPlaneConfiguration.Endpoint.Host),
+	}
+
+	kcp.Spec.KubeadmConfigSpec.PostKubeadmCommands = []string{
+		fmt.Sprintf("/etc/eks/bootstrap-after.sh %s %s", clusterSpec.VersionsBundle.Snow.KubeVip.VersionedImage(), clusterSpec.Spec.ControlPlaneConfiguration.Endpoint.Host),
+	}
+
+	return kcp
+}
+
+func kubeadmConfigTemplate(clusterSpec *cluster.Spec, workerNodeGroupConfig v1alpha1.WorkerNodeGroupConfiguration) bootstrapv1.KubeadmConfigTemplate {
+	kct := clusterapi.KubeadmConfigTemplate(clusterSpec, workerNodeGroupConfig)
+	kct.Spec.Template.Spec.JoinConfiguration.NodeRegistration.KubeletExtraArgs = map[string]string{
```

Comment:
here maybe we should just append to the map instead of overwriting 
Things like oidc args will be shared and probably should be set from `clusterapi.KubeadmControlPlane()`